src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VtableIndexStubGenerator.cs

```diff
@@ -288,48 +354,71 @@ private static IncrementalStubGenerationContext CalculateStubInformation(MethodD
                 virtualMethodIndexData,
+                exceptionMarshallingInfo,
                 ComInterfaceGeneratorHelpers.CreateGeneratorFactory(environment, MarshalDirection.ManagedToUnmanaged),
                 ComInterfaceGeneratorHelpers.CreateGeneratorFactory(environment, MarshalDirection.UnmanagedToManaged),
                 typeKeyType,
                 typeKeyOwner,
                 new SequenceEqualImmutableArray<Diagnostic>(generatorDiagnostics.Diagnostics.ToImmutableArray()));
         }
 
-        private static IMarshallingGeneratorFactory GetMarshallingGeneratorFactory(StubEnvironment env)
+        private static MarshallingInfo CreateExceptionMarshallingInfo(AttributeData virtualMethodIndexAttr, ISymbol symbol, Compilation compilation, GeneratorDiagnostics diagnostics, VirtualMethodIndexData virtualMethodIndexData)
         {
-            IAssemblySymbol coreLibraryAssembly = env.Compilation.GetSpecialType(SpecialType.System_Object).ContainingAssembly;
-            ITypeSymbol? disabledRuntimeMarshallingAttributeType = coreLibraryAssembly.GetTypeByMetadataName(TypeNames.System_Runtime_CompilerServices_DisableRuntimeMarshallingAttribute);
-            bool runtimeMarshallingDisabled = disabledRuntimeMarshallingAttributeType is not null
-                && env.Compilation.Assembly.GetAttributes().Any(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, disabledRuntimeMarshallingAttributeType));
-            InteropGenerationOptions options = new(UseMarshalType: true);
-            IMarshallingGeneratorFactory generatorFactory;
-
-            generatorFactory = new UnsupportedMarshallingFactory();
-            generatorFactory = new NoMarshallingInfoErrorMarshallingFactory(generatorFactory);
-            generatorFactory = new MarshalAsMarshallingGeneratorFactory(options, generatorFactory);
-
-            IMarshallingGeneratorFactory elementFactory = new AttributedMarshallingModelGeneratorFactory(
-                new CharMarshallingGeneratorFactory(generatorFactory, useBlittableMarshallerForUtf16: true), new AttributedMarshallingModelOptions(runtimeMarshallingDisabled, MarshalMode.ElementIn, MarshalMode.ElementRef, MarshalMode.ElementOut));
-            // We don't need to include the later generator factories for collection elements
-            // as the later generator factories only apply to parameters.
-            generatorFactory = new AttributedMarshallingModelGeneratorFactory(generatorFactory, elementFactory, new AttributedMarshallingModelOptions(runtimeMarshallingDisabled, MarshalMode.ManagedToUnmanagedIn, MarshalMode.ManagedToUnmanagedRef, MarshalMode.ManagedToUnmanagedOut));
-
-            generatorFactory = new ByValueContentsMarshalKindValidator(generatorFactory);
-            return generatorFactory;
+            if (!virtualMethodIndexData.ExceptionMarshallingDefined)
```

Comment:
This means we don't respect `ExceptionMarshallingCustomType` if `ExceptionMarshalling` is not specified, right? Is that intentional? I think that would be inconsistent the doc and what we did for `StringMarshalling[CustomType]`.